r/R/get_metadata_information.R

```diff
@@ -30,26 +14,33 @@ getDoubletScore <- function(req, data) {
 # the MT-genes only in MMusculus and Homo Sapiens by grepping "MT"
 #' @export
 getMitochondrialContent <- function(req, data) {
+  result <- formatMetadataResult(data, "percent.mt")
+  return(result)
+}
 
-    # Check if the experiment has percent.mt stored in rds file
-    if(!"percent.mt"%in%colnames(data@meta.data)){
-        stop("MT content is not computed for this experiment")
-    }
+formatMetadataResult <- function(data, column) {
 
-    # Subset the percent.mt ordering by cells_id (DESC)
-    result <- data@meta.data[order(data$cells_id, decreasing = F), "percent.mt"]
-    result<- as.data.frame(result)
-    result$cells_id <- data@meta.data$cells_id[order(data$cells_id, decreasing = F)]
-    result <- result %>% tidyr::complete(cells_id = seq(0,max(data@meta.data$cells_id))) %>% select(-cells_id)
-    result <- t(unname(result))
-    result <- purrr::map(result, function(x) { if(is.na(x)) { return(NULL) } else { return(x) } } )
+  # check if the experiment has specified column
+  if(!column %in% colnames(data@meta.data)){
+    stop(column, " is not computed for this experiment.")
+  }
 
-    # Be aware of possible na values
-    if(any(is.na(result)))
-        warning("There are missing values in the doublet_scores results")
+  # get the specified column, ordering by cells_id
+  cells_id_order <- order(data$cells_id, decreasing = FALSE)
+  result <- data@meta.data[cells_id_order, column]
+  result<- as.data.frame(result)
+  result$cells_id <- data@meta.data$cells_id[cells_id_order]
+  result <- result %>%
+    tidyr::complete(cells_id = seq(0, max(data@meta.data$cells_id))) %>%
+    select(-cells_id)
+  result <- t(unname(result))
```

Comment:
this throws warnings (18 of them) when running tests. `The value argument of names<- can't be empty as of tibble 3.0.0. Column 1 must be named.`

If we care about this, we could "declass" to `"data.frame"` using `as.data.frame` or `class(result) <- "data.frame"`

```suggestion
  result <- t(unname(as.data.frame(result)))
```
If we don't, just leave as is.